### PR TITLE
Merge the same items and refine a list of alternatives in error messages

### PR DESCRIPTION
**Pull request by 23Skidoo**
An updated version of #3776.

### CODE REVIEW

**Comment by ezyang** [comment]
OK. Do we have sample output anywhere?

**Comment by 23Skidoo** [comment]
@ezyang See https://github.com/haskell/cabal/pull/3776#issuecomment-244603838.

**Comment by 23Skidoo** [comment]
Another output example. Old:
```
cabal: Unknown build target 'cabl-install'.
There is no component 'cabl-install' or component 'cabl-install'.
Perhaps you meant the component Cabal or unit-tests or package-tests or Cabal
or unit-tests or package-tests or Cabal or unit-tests or package-tests or
cabal
The project has no package 'cabl-install'.
Perhaps you meant the package Cabal or Cabal or Cabal or cabal-install or
hackage-security or happy or pretty-show or pretty-show
```
New:
```
cabal: Unknown build target 'cabl-install'.
There is no component 'cabl-install'.
The project has no package 'cabl-install'.
Perhaps you meant the package cabal-install
```

**Comment by BardurArantsson** [comment]
Just a minor nitpick. Maybe change

> Perhaps you meant the package cabal-install

to

> Perhaps you meant the package **'**cabal-install**'?**

Note the quotation marks and the question mark.

**Comment by dcoutts** [comment]
Code looks fine.

@BardurArantsson yes please :-)

@phadej agree, should relax to major ver

File: cabal-install/cabal-install.cabal [hunk 1]
```diff
@@ -400,6 +400,7 @@ executable cabal
         containers >= 0.4      && < 0.6,
         cryptohash-sha256 >= 0.11 && < 0.12,
         deepseq    >= 1.3      && < 1.5,
+        edit-distance >= 0.2.2 && < 0.2.3,
```
Comment (23Skidoo) [inline cabal-install/cabal-install.cabal]: Relaxed (in my dev branch).

**Comment by 23Skidoo** [comment]
https://github.com/haskell/cabal/pull/3969#issuecomment-252794525 is the only issue blocking merging right now. Will fix later today or tomorrow.

**Comment by ezyang** [comment]
Obsoleted by #4037.

**Pull request closed without merging**